src/main/js/field/FieldsPanel.js

```diff
@@ -2,27 +2,116 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import * as FieldModelUtilities from 'util/fieldModelUtilities';
 import * as FieldMapping from 'util/fieldMapping';
+import CollapsiblePane from 'component/common/CollapsiblePane';
+
+const DEFAULT_PANEL = 'default';
 
 class FieldsPanel extends React.Component {
     constructor(props) {
         super(props);
+
+        this.initializeFieldMapping = this.initializeFieldMapping.bind(this);
+        this.parsePanel = this.parsePanel.bind(this);
+        this.parseHeader = this.parseHeader.bind(this);
+        this.createDefaultPanel = this.createDefaultPanel.bind(this);
+        this.createPanel = this.createPanel.bind(this);
+        this.createHeaders = this.createHeaders.bind(this);
+        this.createFields = this.createFields.bind(this);
     }
 
-    render() {
+    initializeFieldMapping(fields) {
+        const fieldMapping = {};
+
+        Object.keys(fields).forEach((key) => {
+            const field = fields[key];
+            const { panel, header } = field;
+
+            const panelName = panel || DEFAULT_PANEL;
+            const foundPanel = this.parsePanel(panelName, header, field, fieldMapping);
+            Object.assign(fieldMapping, { [panelName]: foundPanel });
+        });
+
+        return fieldMapping;
+    }
+
+    parsePanel(panelName, headerName, field, fieldMapping) {
+        let foundPanel = fieldMapping[panelName];
```

Comment:
The linter may complain about this.  Might expect 
const foundPanel = fieldMapping[panelName];